anna-sala/pre-curso/pasapalabra.js

```diff
@@ -0,0 +1,171 @@
+                    response = prompt(questions[i].question);
+                    if (response.toLowerCase() === questions[i].answer){
+                        points++;
+                        questions[i].status = 1;
+                        console.log("¡CORRECTO!")
+                    }
+                    else if (response.toLowerCase() === "pasapalabra" || response.toLowerCase() === "pasapalabra "){
+                        console.log("¡Seguimos!")
+                    }
+                    else if (response.toLowerCase() === "end"){
+                        console.log("Has terminado el juego!")
+                        showAnswers();
+                        break;
+                    }
+                    else{
+                        questions[i].status = 3;
+                        incorrectAnswers++;
+                        console.log("¡INCORRECTO!")
+                        alert("La respuesta correcta era: " + questions[i].answer);
+                    }
+                }
+                else if (firstTime===false && questions[i].status===0) {//juego con segunda pregunta
+                    response = prompt(questions[i].secondQuestion);
+                    if (response.toLowerCase() === questions[i].answer){
+                        points++;
+                        questions[i].status = 1;
+                        console.log("¡CORRECTO!")
+                    }
+                    else if (response.toLowerCase() === "pasapalabra" || response.toLowerCase() === "pasapalabra "){
+                        console.log("¡Seguimos!")
+                    }
+                    else if (response.toLowerCase() === "end"){
+                        console.log("Has terminado el juego!")
+                        showAnswers();
+                        break;
+                    }
+                    else{
+                        questions[i].status = 3;
+                        incorrectAnswers++;
+                        console.log("¡INCORRECTO!")
+                        alert("La respuesta correcta era: " + questions[i].answer);
+                    }
+                }
+            }
+       }while(questions.some(newQuestion=> newQuestion.status===0))
+       showAnswers();
+}
+
+function showAnswers(){
+    console.log("Has respondido correctamente a " + points + " preguntas");
+    console.log("Has respondido incorrectamente a " + incorrectAnswers + " preguntas");
+        newObject.name=userName;
+        newObject.points=points;
+        ranking.push(newObject);
+    if (response.toLowerCase() !== "end"){
+        seeRanking();
+    }
+}
+
+function seeRanking(){
+    sortRanking();
+        console.log("El ranking final es:")
+            for(let h=0; h<ranking.length; h++){
+                console.log([h+1] + " - " + ranking[h].name + " con " + ranking[h].points + " puntos! ");
+            }  
+            function sortRanking() 
```

Comment:
Don't declare functions inside others. Refactor this function by putting out sortRanking() and calling it at line 158 instead of declaring it there.